.github/workflows/alert-on-failed-automerge.yml

```diff
@@ -0,0 +1,49 @@
+name: Check Suite Failure Notification
+
+on:
+    # whenever a workflow suite completes trigger this
+    check_suite:
+        types:
+            - completed
+
+jobs:
+    notify_on_failure:
+        if: ${{ github.event.check_suite.conclusion == 'failure' }}
+        runs-on: ubuntu-latest
+
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v3
+
+            - name: Check if the PR has "automerge" label
+              id: automerge_check
+              run: |
+                  pr_number=$(jq -r '.check_suite.pull_requests[0].number' <<< "${{ toJson(github.event) }}")
+                  if [ -z "$pr_number" ]; then
+                    echo "No PR associated with this check suite."
+                    echo "::set-output name=result::false"
+                    exit 0
+                  fi
+
+                  labels=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+                  https://api.github.com/repos/${{ github.repository }}/issues/$pr_number/labels)
+
+                  echo "Labels: $labels"
+                  if echo "$labels" | grep -q "automerge"; then
+                    echo "::set-output name=result::true"
+                  else
+                    echo "::set-output name=result::false"
+                  fi
+
+            - name: Send Slack notification if the PR has "automerge" label
+              if: ${{ steps.automerge_check.outputs.result == 'true' }}
+              uses: rtCamp/action-slack-notify@v2
+              env:
+                  SLACK_CHANNEL: alerts
```

Comment:
Could this be noisy if automerged PRs are going to a widely used channel?

Reply:
yep! that's what i want :D

this generally works fine so we get lulled into a false sense of security, and then it fails and we miss it, personally sometimes for hours because I'm always either balancing multiple bits of work, or covered in kids

so it should be quiet - and then periodically noisy - can always tune if we e.g. got 10 notifications for 1 logical instance of the problem

Reply:
but fair that it shouldn't be there while we're testing it!